src/shared/Microsoft.Git.CredentialManager/UriExtensions.cs

```diff
@@ -2,12 +2,29 @@
 // Licensed under the MIT license.
 using System;
 using System.Collections.Generic;
+using System.Linq;
 using System.Net;
+using System.Web;
 
 namespace Microsoft.Git.CredentialManager
 {
     public static class UriExtensions
     {
+        public static IDictionary<string, string> GetQueryParameters(this Uri uri)
+        {
+            return ParseQueryString(uri.Query);
+        }
+        
+        private static IDictionary<string, string> ParseQueryString(string query)
+        {
+            return query
+                .TrimStart('?')
+                .Split('&')
+                .ToDictionary(
+                    x => x.Split('=')[0],
+                    x => HttpUtility.HtmlDecode(x.Split('=')[1]));
```

Comment:
Why HtmlDecode and not UrlDecode if we're working with a query string?

Reply:
Whoops!